src/source-dpdk.c

```diff
@@ -0,0 +1,494 @@
+                SCReturnInt(EXIT_FAILURE);
+            }
+            PKT_SET_SRC(p, PKT_SRC_WIRE);
+            p->datalink = LINKTYPE_ETHERNET;
+            if (ptv->checksum_mode == CHECKSUM_VALIDATION_DISABLE) {
+                p->flags |= PKT_IGNORE_CHECKSUM;
+            }
+
+            DPDKSetTimeval(&p->ts);
+            p->dpdk_v.mbuf = ptv->received_mbufs[i];
+            p->ReleasePacket = DPDKReleasePacket;
+            p->dpdk_v.copy_mode = ptv->copy_mode;
+            p->dpdk_v.out_port_id = ptv->out_port_id;
+            p->dpdk_v.out_queue_id = ptv->queue_id;
+
+            if (PacketSetData(p, rte_pktmbuf_mtod(p->dpdk_v.mbuf, uint8_t *),
+                        rte_pktmbuf_pkt_len(p->dpdk_v.mbuf)) == -1) {
+                TmqhOutputPacketpool(ptv->tv, p);
+                DPDKFreeMbufArray(ptv->received_mbufs, nb_rx - i - 1, i + 1);
+                SCReturnInt(EXIT_FAILURE);
+            }
+
+            if (TmThreadsSlotProcessPkt(ptv->tv, ptv->slot, p) != TM_ECODE_OK) {
+                TmqhOutputPacketpool(ptv->tv, p);
+                DPDKFreeMbufArray(ptv->received_mbufs, nb_rx - i - 1, i + 1);
+                SCReturnInt(EXIT_FAILURE);
+            }
+        }
+
+        /* Trigger one dump of stats every second */
+        current_time = DPDKGetSeconds();
+        if (current_time != last_dump) {
+            DPDKDumpCounters(ptv);
+            last_dump = current_time;
+        }
+    }
+
+    SCReturnInt(TM_ECODE_OK);
+}
+
+/**
+ * \brief Init function for ReceiveDPDK.
+ *
+ * \param tv pointer to ThreadVars
+ * \param initdata pointer to the interface passed from the user
+ * \param data pointer gets populated with DPDKThreadVars
+ *
+ */
+TmEcode ReceiveDPDKThreadInit(ThreadVars *tv, const void *initdata, void **data)
+{
+    SCEnter();
+    int retval;
+    DPDKIfaceConfig *dpdk_config = (DPDKIfaceConfig *)initdata;
+
+    if (initdata == NULL) {
+        SCLogError(SC_ERR_INVALID_ARGUMENT, "DPDK Configuration failed");
+        dpdk_config->DerefFunc(dpdk_config);
+        SCReturnInt(TM_ECODE_FAILED);
+    }
+
+    DPDKThreadVars *ptv = SCCalloc(1, sizeof(DPDKThreadVars));
+    if (unlikely(ptv == NULL)) {
+        SCLogError(SC_ERR_MEM_ALLOC, "Unable to allocate memory");
+        dpdk_config->DerefFunc(dpdk_config);
+        SCReturnInt(TM_ECODE_FAILED);
+    }
+
+    ptv->tv = tv;
+    ptv->pkts = 0;
+    ptv->bytes = 0;
+    ptv->livedev = LiveGetDevice(dpdk_config->iface);
+
+#ifdef PACKET_STATISTICS
+    ptv->capture_dpdk_packets = StatsRegisterCounter("capture.packets", ptv->tv);
+    ptv->capture_dpdk_drops = StatsRegisterCounter("capture.drops", ptv->tv);
+#endif
+
+    ptv->copy_mode = dpdk_config->copy_mode;
+    ptv->checksum_mode = dpdk_config->checksum_mode;
+
+    ptv->threads = dpdk_config->threads;
+    ptv->port_id = dpdk_config->port_id;
+    ptv->out_port_id = dpdk_config->out_port_id;
+    uint16_t queue_id = SC_ATOMIC_ADD(dpdk_config->queue_id, 1);
+    ptv->queue_id = queue_id;
```

Comment:
please refer YAML comment on TX queue on copy port. rework

Reply:
but its DPDK port_init logic to inform, warn or graceful exit if config is incorrect.

Reply:
Added https://github.com/OISF/suricata/pull/6272#discussion_r698658957